src/main/java/de/tum/in/www1/artemis/service/connectors/BitbucketBambooUpdateService.java

```diff
@@ -0,0 +1,112 @@
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.math.NumberUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import com.appfire.bamboo.cli.BambooClient;
+import com.appfire.common.cli.CliClient;
+
+import de.tum.in.www1.artemis.config.Constants;
+import de.tum.in.www1.artemis.exception.BambooException;
+import de.tum.in.www1.artemis.service.connectors.bamboo.BambooBuildPlanUpdateProvider;
+
+@Service
+// Only activate this service bean, if both Bamboo and Bitbucket are activated (@Profile({"bitbucket","bamboo"}) would activate
+// this if any profile is active (OR). We want both (AND)
+@Profile("bamboo & bitbucket")
+public class BitbucketBambooUpdateService implements ContinuousIntegrationUpdateService {
+
+    @Value("${artemis.version-control.url}")
+    private URL BITBUCKET_SERVER;
+
+    private final Logger log = LoggerFactory.getLogger(BitbucketBambooUpdateService.class);
+
+    private final BambooClient bambooClient;
+
+    private final BambooBuildPlanUpdateProvider bambooBuildPlanUpdateProvider;
+
+    public BitbucketBambooUpdateService(BambooClient bambooClient, BambooBuildPlanUpdateProvider bambooBuildPlanUpdateProvider) {
+        this.bambooClient = bambooClient;
+        this.bambooBuildPlanUpdateProvider = bambooBuildPlanUpdateProvider;
+    }
+
+    @Override
+    public void updatePlanRepository(String bambooProject, String planKey, String bambooRepositoryName, String bitbucketProject, String bitbucketRepository,
+            Optional<List<String>> triggeredBy) {
+        try {
+            log.debug("Update plan repository for build plan " + planKey);
+            com.appfire.bamboo.cli.objects.RemoteRepository bambooRemoteRepository = bambooClient.getRepositoryHelper().getRemoteRepository(bambooRepositoryName, planKey, false);
+            // Workaround for old exercises which used a different repositoryName
+            if (bambooRemoteRepository == null) {
+                bambooRemoteRepository = bambooClient.getRepositoryHelper().getRemoteRepository("Assignment", planKey, false);
+                if (bambooRemoteRepository == null) {
+                    throw new BambooException("Something went wrong while updating the template repository of the build plan " + planKey
+                            + " to the student repository : Could not find assignment nor Assignment repository");
+                }
+            }
+
+            bambooBuildPlanUpdateProvider.updateRepository(bambooRemoteRepository, bitbucketRepository, bitbucketProject, planKey);
+
+            // Overwrite triggers if needed, incl workaround for different repo names
+            if (triggeredBy.isPresent() && bambooRemoteRepository.getName().equals("Assignment")) {
```

Comment:
Could the string be replaced by an enum or at least a local variable to not repeat `"Assignment"`?

Reply:
I refactored this into a constant.